ta/crypt/cryp_taf.c

```diff
@@ -47,104 +49,121 @@ TEE_Result ta_entry_allocate_operation(uint32_t param_type, TEE_Param params[4])
 	res = TEE_AllocateOperation(&op,
 				    params[0].value.b, params[1].value.a,
 				    params[1].value.b);
-	params[0].value.a = (uint32_t) op;
+	params[0].value.a = (uintptr_t)op;
```

Comment:
OK, got it, it will work. But it does not make the TA really 64-bit clean. I mean, ideally the TA should not need to assume anything about the value of a `TEE_OperationHandle`. What about reserving both `params[0].value.a` and `params[0].value.b` for the handle? Would it be too intrusive a change?